l5kit/l5kit/tests/simulation/e2e_test.py

```diff
@@ -0,0 +1,122 @@
+import pytest
+import torch
+
+from l5kit.cle.closed_loop_evaluator import ClosedLoopEvaluator, EvaluationPlan
+from l5kit.cle.metrics import (CollisionFrontMetric, CollisionRearMetric, CollisionSideMetric,
+                               DisplacementErrorL2Metric, DistanceToRefTrajectoryMetric)
+from l5kit.cle.validators import RangeValidator, ValidationCountingAggregator
+from l5kit.dataset import EgoDataset
+from l5kit.simulation.dataset import SimulationConfig
+from l5kit.simulation.unroll import ClosedLoopSimulator
+from l5kit.tests.simulation.unroll_test import MockModel
+
+
+@pytest.fixture(scope="function")
+def simulation_evaluator() -> ClosedLoopEvaluator:
+    return ClosedLoopEvaluator(EvaluationPlan(metrics=[DisplacementErrorL2Metric(),
+                                                       DistanceToRefTrajectoryMetric(),
+                                                       CollisionFrontMetric(),
+                                                       CollisionRearMetric(),
+                                                       CollisionSideMetric()],
+                                              validators=[RangeValidator("displacement_error_l2_validator",
+                                                                         DisplacementErrorL2Metric, max_value=30),
+                                                          RangeValidator("distance_ref_trajectory_validator",
+                                                                         DistanceToRefTrajectoryMetric, max_value=4),
+                                                          RangeValidator("collision_front_validator",
+                                                                         CollisionFrontMetric, max_value=0),
+                                                          RangeValidator("collision_rear_validator",
+                                                                         CollisionRearMetric, max_value=0),
+                                                          RangeValidator("collision_side_validator",
+                                                                         CollisionSideMetric, max_value=0),
+                                                          ],
+                                              composite_metrics=[],
+                                              intervention_validators=["displacement_error_l2_validator",
+                                                                       "distance_ref_trajectory_validator",
+                                                                       "collision_front_validator",
+                                                                       "collision_rear_validator",
+                                                                       "collision_side_validator"]))
```

Comment:
You can split it in variables (`metric_list`, `validator_list`, etc) like we have in tests in avjadoo:

```python
        metric_list: Iterable[metrics.SupportsMetricCompute] = [
            metrics.CollisionFrontMetric(),
            metrics.CollisionRearMetric(),
            metrics.CollisionSideMetric(),
            metrics.DisplacementErrorL2Metric(),
            metrics.DistanceToRefTrajectoryMetric(),
        ]

        # Validators for the evaluation plan
        validators_list: Iterable[validators.SupportsMetricValidate] = [
            validators.RangeValidator("front_collision_validator",
                                      metrics.CollisionFrontMetric,
                                      max_value=0),
            validators.RangeValidator("side_collision_validator",
                                      metrics.CollisionSideMetric,
                                      max_value=0),
            validators.RangeValidator("rear_collision_validator",
                                      metrics.CollisionRearMetric,
                                      max_value=0),
            validators.RangeValidator("displacement_error_validator",
                                      metrics.DisplacementErrorL2Metric,
                                      max_value=0),
            validators.RangeValidator("distance_to_ref_trajectory_validator",
                                      metrics.DistanceToRefTrajectoryMetric,
                                      max_value=0)
        ]

        # Builds the evaluation plan and run the closed loop evaluator
        evaluation_plan = ceval.EvaluationPlan(metric_list, validators_list)
        evaluator = ceval.ClosedLoopEvaluator(evaluation_plan)
```

Just to make it easier to read.